benchmark.py

```
@@ -0,0 +1,838 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class BenchmarkConfig:
+    num_docs: int = 10_000
+    num_searches: int = 1_000
+    num_runs: int = 3
+    server_exe: str = 'zig-out/bin/fpindex'
+    server_port: int = 8999
+    index_name: str = 'benchmark_index'
+    data_dir_base: str = '/tmp/fpindex_benchmark'
```

Ruff (0.12.2)

Probable insecure usage of temporary file or directory: "/tmp/fpindex_benchmark"

(S108)